### PR TITLE
Fix grokker rules that have fields with common prefixes

### LINKED ISSUE #564: Grokker won't allow for extension of already existing field
If a grokker rule contains two similiar fields and the second one is the same only extended by "_something", then the field wont get written.

Example: ......... ( [A-Za-z: ]*\%)?(%{CISCO_REASON:cisco.**facility**}......%{CISCO_REASON:cisco.**facility_mnoe**}...................

### PR DESCRIPTION
**Pull request by dtrai2**
closes #564

### CODE REVIEW

File: logprep/processor/grokker/rule.py [hunk 1]
```diff
@@ -68,10 +68,14 @@ def _dotted_field_to_logstash_converter(mapping: dict) -> dict:
     def _transform(pattern):  # nosemgrep
         fields = re.findall(FIELD_PATTERN, pattern)
         for dotted_field, _ in fields:
-            splitted_field = dotted_field.split(".")
-            if len(splitted_field) > 1:
-                replacement = "".join(f"[{element}]" for element in splitted_field)
-                pattern = re.sub(re.escape(dotted_field), replacement, pattern)
+            split_field = dotted_field.split(".")
+            if len(split_field) > 1:
```
Comment (ekneg54) [inline logprep/processor/grokker/rule.py:72]:
```suggestion
            if "." in dotted_field:
```
* if a dot is in dotted field, the length will be greater than 1
* much cheaper operation than two method calls

**Review by ekneg54 — CHANGES_REQUESTED** [review]
thank you for your contribution.
please consider to have a look at one minor change

File: logprep/processor/grokker/rule.py (hunk 1 above)
Comment (dtrai2) [reply logprep/processor/grokker/rule.py:72]: Well we still need the `dotted_field.split(".")` in the next line. So what I could offer is something like following. But it doesn't really reduce the method calls except the change of `len(...) > 1` to `"." in dotted_field`.
```python
if "." in dotted_field:
    replacement = "".join(f"[{element}]" for element in dotted_field.split("."))
```

Comment (ekneg54) [reply logprep/processor/grokker/rule.py:72]: still better 👍

**Review by ekneg54 — APPROVED** [review]

**Pull request merged**